slice.go

```diff
@@ -131,4 +139,142 @@ func(s *slice[E]) Pop() (E, bool) {
 	(*s)[n-1] = empty
 	*s = (*s)[:n-1]
 	return v, true
-}
\ No newline at end of file
+}
+
+func (s Slice[E]) Copy() Slice[E] {
+	var cp = make(Slice[E], s.Len())
+	copy(cp, s)
+	return cp
+}
+
+func (s Slice[E]) CopyWith(fn func(E) E) Slice[E] {
```

Comment:
What is the idea behind `fn` ? I mean when and why we need `CopyWith`

Reply:
To use user defined deep copy